Minesweeper/Minesweeper/Entities/MineField.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  MineField.swift
+//  Minesweeper
+//
+//  Created by Dorian Grolaux on 19/02/2018.
+//
+
+import Foundation
+
+typealias Coordinates = (x: Int, y: Int)
+
+class MineField {
+    private let cells: [[Cell]]!
+    
+    init(cells: [[Cell]]) {
+        self.cells = cells
+    }
+    
+    
+    init(gridSize size: Int, withBombAmount bombs: Int) {
+        
+        let randomPlacements = MineField.getPlacementOptions(count: bombs, forArraySize: size)
+        var cells = [[Cell]]()
+        
+        for x in 0..<size {
+            var yCells = [Cell]()
+            
+            for y in 0..<size {
+                yCells.append(Cell(containsBomb: randomPlacements.contains(where: { placement -> Bool in
+                    return placement == (x: x, y: y)
+                })))
+            }
+            cells.append(yCells)
+        }
+        
+        MineField.markPlacedBombs(inField: &cells, placedBombs: randomPlacements)
+        
+        self.cells = cells
+    }
+    
+    private static func getImmediateNeighbours(ofCellAt cellCoords: Coordinates, inField field: [[Cell]]) -> [Coordinates] {
```

Comment:
I'm wondering about the best way to do this.

Is it the grids responsibility to do a one time check or is it the cells responsibility to know who it's neighbours are? Genuine question, I'm not sure.

Reply:
I think it is, the grid manages the cell placement, and cells have no knowledge of their neighbours (and they don't need to know about them either, or at least not in any situation that I can think of in the context of this game)